### PR TITLE
Do not return HSI attributes when running in a VM

### PR DESCRIPTION
**Pull request by hughsie**

### CODE REVIEW

File: src/fu-main.c [hunk 1]
```diff
@@ -1662,6 +1670,16 @@ fu_main_load_introspection (const gchar *filename, GError **error)
...
 }
 
+static gboolean
+fu_main_is_hypervisor (void)
+{
+	g_autofree gchar *buf = NULL;
+	gsize bufsz = 0;
+	if (!g_file_get_contents ("/proc/cpuinfo", &buf, &bufsz, NULL))
+		return FALSE;
+	return g_strstr_len (buf, (gssize) bufsz, "hypervisor") != NULL;
```
Comment (superm1) [inline src/fu-main.c:1680]: Is this valid for all VMs?  Or does there need to be more ways to detect?

File: src/fu-main.c [hunk 2]
```diff
@@ -1105,6 +1106,13 @@ fu_main_daemon_method_call (GDBusConnection *connection, const gchar *sender,
 	if (g_strcmp0 (method_name, "GetHostSecurityAttrs") == 0) {
 		g_autoptr(FuSecurityAttrs) attrs = NULL;
 		g_debug ("Called %s()", method_name);
+		if (priv->is_vm) {
+			g_dbus_method_invocation_return_error_literal (invocation,
+								       FWUPD_ERROR,
+								       FWUPD_ERROR_NOT_SUPPORTED,
+								       "HSI unavailable for hypervisor");
```
Comment (superm1) [inline src/fu-main.c:1113]: I think the same thing should probably apply for containers too.

File: src/fu-main.c (hunk 1 above)
Comment (hughsie) [reply src/fu-main.c:1680]: I think the kernel has a pretty long list of ways to detect a VM.

File: src/fu-main.c (hunk 2 above)
Comment (hughsie) [reply src/fu-main.c:1113]: Oof, that's harder. Do you know how to detect that?

Comment (superm1) [reply src/fu-main.c:1113]: https://stackoverflow.com/questions/20010199/how-to-determine-if-a-process-runs-inside-lxc-docker seems to have a few ways to do it

Comment (exploide) [reply src/fu-main.c:1113]: `systemd-detect-virt` has also implemented such checks: https://github.com/systemd/systemd/blob/5b14956385ee1076b98c3ca730bb9c1c9e95fc22/src/basic/virt.c#L450

Comment (hughsie) [reply src/fu-main.c:1119]: I've added something that should catch 99% of people trying this.

**Pull request merged**